src/color.rs

```diff
@@ -682,6 +688,57 @@ where
     }
 }
 
+impl<S: Primitive, T: Primitive> FromColor<Cmyk<S>> for Rgb<T>
+where
+    T: FromPrimitive<S>,
+{
+    fn from_color(&mut self, other: &Cmyk<S>) {
+        let [r, g, b] = &mut self.0;
+
+        (*r, *g, *b) = cmyk_to_rgb(other.0);
+    }
+}
+
+fn cmyk_to_rgb<S: Primitive, T: Primitive>([c, m, y, k]: [S; 4]) -> (T, T, T) {
```

Comment:
That's the best I could come up with. The intermediate conversion to f32 seems weird, but I also don't know how to avoid it.